Clean up screen path code

IanSav opened the pull request:
[Screen.py] Remove defunct "screen_path"
- All official OpenPLi skins have now been updated to use the newer "ScreenPath" widget so the defunct "screen_path" widget can now be removed.

[UsageConfig.py] Transition screen path settings
- This temporary code will find the user's current value for the old "config.usage.menu_path" setting and move it to the new "config.usage.showScreenPath".  The code will then remove the old "config.usage.menu_path" entry from the settings file.

- This code is only run once such that when the setting is transferred the old setting is no longer used.  It is intended to remove this code after one or two production releases of OpenPLi when it will be assumed that most users have had the setting transferred.  When this code is removed the only implication for users will be that the setting will not be migrated and the "config.usage.showScreenPath" default value will be used.  Also the old "config.usage.menu_path" setting, if not at default, will not be removed.

[setup.xml] Use the new showScreenPath setting
- This change uses the new "config.usage.showScreenPath" setting and improves the label and description text to better reflect the functionality.

[Screen.py] Use new "config.usage.showScreenPath"
- This change replaces the old "config.usage.menu_path" setting with the new and proposed common "config.usage.showScreenPath" setting.

[UsageConfig.py] Remove transition code
- This change was requested by Littlesat and Persian Prince as OpenPLi production releases are too far apart such that this code could hang around for a very long time.  It is not considered an issue for this setting to be reset to default and the old value to be left in the setting file.

IanSav committed:
[Screen.py] Remove defunct "screen_path"

All official OpenPLi skins have now been updated to use the newer "ScreenPath" widget so the defunct "screen_path" widget can now be removed.

IanSav committed:
[UsageConfig.py] Transition screen path settings

This temporary code will find the user's current value for the old "config.usage.menu_path" setting and move it to the new "config.usage.showScreenPath".  The code will then remove the old "config.usage.menu_path" entry from the settings file.

This code is only run once such that when the setting is transferred the old setting is no longer used.  It is intended to remove this code after one or two production releases of OpenPLi when it will be assumed that most users have had the setting transferred.  When this code is removed the only implication for users will be that the setting will not be migrated and the "config.usage.showScreenPath" default value will be used.  Also the old "config.usage.menu_path" setting, if not at default, will not be removed.

IanSav committed:
[setup.xml] Use the new showScreenPath setting

This change uses the new "config.usage.showScreenPath" setting and improves the label and description text to better reflect the functionality.

IanSav committed:
[Screen.py] Use new "config.usage.showScreenPath"

This change replaces the old "config.usage.menu_path" setting with the new and proposed common "config.usage.showScreenPath" setting.

littlesat commented:
I suggest we remove the temporary code.... We can put it at the release notes that user need to reconfigure this setting... I just suggest to change the default to small instead.

IanSav commented:
I would prefer that we try to rescue the settings for users.  It will make my change less disruptive.  I will ensure this code is removed after update after the next update.  This code will *not* last long.  It only ever runs once and will clean up the user's setting file.  Without this code the settings file may retain a stale entry.

persianpros commented:
develop branch won't be used for normal releases so users might see this for example 1 month later, then we need to wait till then so it's better to not have it :+1:

IanSav commented:
So how can we clean up the old setting value?

IanSav committed:
[UsageConfig.py] Remove transition code

This change was requested by Littlesat and Persian Prince as OpenPLi production releases are too far apart such that this code could hang around for a very long time.  It is not considered an issue for this setting to be reset to default and the old value to be left in the setting file.

IanSav commented:
Pull request updated as requested.
